src/state/selectors/axisSelectors.ts

```diff
@@ -1683,8 +1683,9 @@ export const combineAxisTicks = (
     return result.filter((row: TickItem) => !isNan(row.coordinate));
   }
 
-  // When axis is a categorical axis, but the type of axis is number or the scale of axis is not "auto"
-  if (isCategorical && categoricalDomain) {
+  // When axis is a categorical axis, but the type of axis is number or the scale of axis is not "auto".
+  // Unless the scale is a time scale or a custom scale, in which case defer to the scale.ticks function below.
+  if (isCategorical && categoricalDomain && axis.scale !== 'time' && typeof axis.scale !== 'function') {
```

Comment:
I think this is a _good_ change, but I don't want this particular function to become this place where start adding more conditions in order to avoid the below block of code.

I'm not sure if there are more scale types within d3-scale that would also fall into this bucket, but if there are I'd rather not list them out here.

Will let @PavelVanecek review and see if he has any ideas